cairo/src/contracts/token/extensions/hyp_erc721_URI_storage.cairo

```diff
@@ -1,48 +1,226 @@
+        mailbox: ContractAddress,
+        _mint_amount: u256,
+        _name: ByteArray,
+        _symbol: ByteArray,
+        _base_uri: ByteArray,
+        _hook: ContractAddress,
+        _interchainSecurityModule: ContractAddress,
+    ) {
+        self.ownable.initializer(owner);
+        self.mailboxclient.initialize(mailbox, Option::Some(_hook), Option::Some(_interchainSecurityModule));
+        self.hyp_erc721.initialize(
+            _mint_amount,
+            _name,
+            _symbol,
+            _base_uri
+        )
+    }
 
-        fn balance_of(self: @ContractState, account: u256) -> u256 {
-            0
+    #[abi(embed_v0)]
+    impl HypErc721Upgradeable of IUpgradeable<ContractState> {
+        fn upgrade(ref self: ContractState, new_class_hash: starknet::ClassHash) {
+            self.ownable.assert_only_owner();
+            self.upgradeable.upgrade(new_class_hash);
         }
+    }
 
-        fn token_uri(self: @ContractState, token_id: u256) -> u256 {
-            0
-        }
+    // #[generate_trait]
+    // impl InternalImpl of InternalTrait {
+    //     fn transfer_from_sender(ref self: ContractState, token_id: u256) -> u256 {
+    //         self.hyp_erc721.transfer_from_sender(token_id);
+    //         token_id
+    //     }
 
-        fn supports_interface(self: @ContractState, interface_id: u256) -> bool {
-            false
-        }
-    }
+    //     fn transfer_to(ref self: ContractState, recipient: ContractAddress, token_id: u256, token_uri: u256) {
+    //         self.hyp_erc721.transfer_to(recipient, token_id);
+    //     }
 
-    #[generate_trait]
-    impl InternalImpl of InternalTrait {
-        fn transfer_from_sender(ref self: ContractState, token_id: u256) -> u256 {
-            0
-        }
+    //     fn before_token_transfer(
+    //         ref self: ContractState, from: u256, to: ContractAddress, token_id: u256, batch_size: u256
+    //     ) {
+    //         self.erc721.before_update(to, token_id, Zero::zero());
+    //     }
+    // }
 
-        fn transfer_to(ref self: ContractState, recipient: u256, token_id: u256, token_uri: u256) {}
+    // would be extended when erc721_enumerable is imported
+    impl ERC721HooksImpl of ERC721HooksTrait<ContractState> {
+        fn before_update(
+            ref self: ERC721Component::ComponentState<ContractState>,
+            to: ContractAddress,
+            token_id: u256,
+            auth: ContractAddress
+        ) {}
 
-        fn before_token_transfer(
-            ref self: ContractState, from: u256, to: u256, token_id: u256, batch_size: u256
+        fn after_update(
+            ref self: ERC721Component::ComponentState<ContractState>,
+            to: ContractAddress,
+            token_id: u256,
+            auth: ContractAddress
         ) {}
+    }
 
-        fn burn(ref self: ContractState, token_id: u256) {}
+
+    impl TokenRouterHooksImpl of TokenRouterHooksTrait<ContractState> {
+        fn transfer_from_sender_hook(
```

Comment:
This does not look right, it call `self.hyp_erc721.transfer_from_sender()` and return the token_uri as `Bytes`